### DISCUSSION TITLE
partial search terms with DB PostgresSQL

### DISCUSSION THREAD

**Discussion by dvviktordelev** [requarks/wiki#6444] [Help / Questions]
Hello,
when searching for e.g "data" it doesn't show any results (although "database" content exists in our wiki). Searching for "database" or "data*" shows "database" entries. Is this normal behavior? Is there a way to change this?

Any help is greatly appreciated.

Best,
Viktor

**Comment by ltjessem** [comment]
Tbh the PostgreSQL search is horrible. It reminds me of the ever frustrating Windows search. Here's an example from my wiki:

Inte - 4 results, but not the one I'm looking for.
Integ - No results
Integr - One result
Integra - No results
Integrat - No results
Integrati - No results
Integratio - No results
Integration - One result

**Comment by thinkh** [comment]
Is it possible to create custom search provider in wiki.js? If yes, one could implement a search provider for https://github.com/meilisearch/meilisearch. It supports also JavaScript and is open source.

**Reply by thinkh** [reply]
Creating a search engine module shouldn't be hard according to the [wiki.js docs](https://docs.requarks.io/dev/search) and [meilisearch docs](https://www.meilisearch.com/docs/learn/getting_started/quick_start).

**Reply by thinkh** [reply]
Alternatively, one could switch to elasticsearch (either cloud or self-hosted via docker).

And if one is open to try something advanced, one could feed the documents into [haystack](https://haystack.deepset.ai). Haystack implements production-ready semantic search, question answering, summarization and document ranking. It also [stores the documents in elasticsearch and connected via REST API](https://haystack.deepset.ai/tutorials/20_using_haystack_with_rest_api). The result set from the search looks similar to the elasticsearch one. So maybe one could extend or reuse the existing wiki.js search module for this case.
